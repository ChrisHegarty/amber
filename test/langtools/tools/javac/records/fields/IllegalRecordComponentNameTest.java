/*
 * @test /nodynamiccopyright/
 * @summary dont accept illegal record component names
 * @compile/fail/ref=IllegalRecordComponentNameTest.out -XDrawDiagnostics IllegalRecordComponentNameTest.java
 */

public class IllegalRecordComponentNameTest {
    record R1(String toString) {}
    record R2(String hashCode) {}
    record R3(String getClass) {}
    record R4(String readObjectNoData) {}
    record R5(String readResolve) {}
    record R6(String writeReplace) {}
    record R7(String serialPersistentFields) {}
}
