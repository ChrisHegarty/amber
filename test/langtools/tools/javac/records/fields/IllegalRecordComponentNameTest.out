IllegalRecordComponentNameTest.java:8:22: compiler.err.illegal.record.component.name: toString
IllegalRecordComponentNameTest.java:9:22: compiler.err.illegal.record.component.name: hashCode
IllegalRecordComponentNameTest.java:10:22: compiler.err.illegal.record.component.name: getClass
IllegalRecordComponentNameTest.java:11:22: compiler.err.illegal.record.component.name: readObjectNoData
IllegalRecordComponentNameTest.java:12:22: compiler.err.illegal.record.component.name: readResolve
IllegalRecordComponentNameTest.java:13:22: compiler.err.illegal.record.component.name: writeReplace
IllegalRecordComponentNameTest.java:14:22: compiler.err.illegal.record.component.name: serialPersistentFields
7 errors
